pyartifactory/objects/build.py

```diff
@@ -0,0 +1,206 @@
+            _del = self._post(f"api/{self._uri}/delete", json=delete_build.model_dump())
+            logger.debug(_del.text)
+
+    def build_rename(self, build_name: str, new_build_name: str) -> None:
+        """
+        :param build_name: Build to be renamed
+        :param new_build_name: New Build name
+        :return: None
+        """
+        try:
+            response = self._get(
+                f"api/{self._uri}/{build_name}",
+            )
+            self._response_checker(response)
+        except requests.exceptions.HTTPError as error:
+            http_response: Union[Response, None] = error.response
+            if isinstance(http_response, Response) and http_response.status_code == 404:
+                raise BuildNotFoundError(f"Build {build_name} were not found")
+            raise ArtifactoryError from error
+        except ArtifactoryBuildError as error:
+            raise ArtifactoryError from error
+        else:
+            self._post(f"api/{self._uri}/rename/{build_name}?to={new_build_name}")
+            logger.debug("Build %s successfully renamed to %s", build_name, new_build_name)
+
+    def build_diff(self, build_name: str, build_number: str, older_build_number: str) -> BuildDiffResponse:
+        """
+        :param build_name: Build name to be compared
+        :param build_number: More recent build to be compared
+        :param older_build_number: Starting build to be compared
+        :return: BuildDiffResponse model object containing server response
+        """
+        build_name = build_name.lstrip("/")
+        build_number = build_number.lstrip("/")
+        older_build_number = older_build_number.lstrip("/")
+        # self._check_build_number_value(build_number)
+        # self._check_build_number_value(older_build_number)
+
+        try:
+            response = self._get(
+                f"api/{self._uri}/{build_name}/{build_number}?diff={older_build_number}",
+            )
+            self._response_checker(response)
+            logger.debug("Build Diff successfully retrieved")
+            return BuildDiffResponse(**response.json())
+        except requests.exceptions.HTTPError as error:
+            http_response: Union[Response, None] = error.response
+            if isinstance(http_response, Response) and http_response.status_code == 404:
+                raise BuildNotFoundError(
+                    f"Build diff {build_number} or {older_build_number} were not found on {build_name}",
+                )
+            raise ArtifactoryError from error
+        except ArtifactoryBuildError as error:
+            raise ArtifactoryError from error
+
+    def _response_checker(self, response):
+        try:
+            _error = BuildError(**response.json())
+        except pydantic_core.ValidationError:
+            # response does not fit with error model
+            return
+        else:
+            raise ArtifactoryBuildError(_error.errors[0].status, _error.errors[0].message)
```

Comment:
You don't need to create your own response checker, there is already one available on the project (see [here](https://github.com/anancarv/python-artifactory/blob/master/pyartifactory/objects/object.py#L91)). Indeed, the API requests are always followed by a `raise_for_status()` method, so there is an implicit response checking.

Reply:
that `raise_for_status()` works when an [HTTPerror happens](https://requests.readthedocs.io/en/latest/api/#requests.Response.raise_for_status). 

I've checked with real JFrog installation that when looking for absent jfrog builds, you receive an http 200 json response (so no http error happens, `raise_for_status()` is useless here) that I've modeled with `BuildError` like 
```json
{
  errors: [  { "status": 404 , "message": "build xxx not found in yyyy." }]
}
```
fields reused when raising ArtifactoryBuildError.

Reply:
Ok, I get it. But still, I think it's better to update the `_generic_http_method_request` instead of creating a new response checker.

Reply:
I would say that you should verify that the response is in the format `requests.Response` in the `_generic_http_method_request` before calling `raise_for_status`. It's exactly what you're doing here, but simpler.

> I've checked with real JFrog installation that when looking for absent jfrog builds, you receive an http 200 json response

By the way, what is the actual response sent back by Artifactory?

Reply:
I'm sorry, I checked it better today and I confirm that HTTPError is raised from `raise_for_status()` .
I've changed some code to handle all exception in same way, using a model to manage the error messages.